README.md

```diff
@@ -82,8 +84,15 @@ The above config will export time series such as
 ```
 # HELP aws_elb_request_count_sum CloudWatch metric AWS/ELB RequestCount Dimensions: ["AvailabilityZone","LoadBalancerName"] Statistic: Sum Unit: Count
 # TYPE aws_elb_request_count_sum gauge
-aws_elb_request_count_sum{job="aws_elb",load_balancer_name="mylb",availability_zone="eu-west-1c",} 42.0
-aws_elb_request_count_sum{job="aws_elb",load_balancer_name="myotherlb",availability_zone="eu-west-1c",} 7.0
+aws_elb_request_count_sum{job="aws_elb",instance="",load_balancer_name="mylb",availability_zone="eu-west-1c",} 42.0
+aws_elb_request_count_sum{job="aws_elb",instance="",load_balancer_name="myotherlb",availability_zone="eu-west-1c",} 7.0
+```
+
+If `enable_aws_resource_info` is set to true, an additional information metric will be exported for each AWS resource matched by `aws_tag_select`, such as
```

Comment:
It should apply to everything.

Reply:
@brian-brazil I know, but as you told the opt-in was "a question of permission", I though you mean the boolean would be enough.

I would have like to apply it to everything, but the dependency to the tag selection is here because the necessary meta data is received through the [GetResource API](https://docs.aws.amazon.com/resourcegroupstagging/latest/APIReference/API_GetResources.html). The API takes type and tag filters, and returns all tags mapped to an ARN; this result can't be generically correlated to the CloudWatch dimensions, as the available dimensions may or may not contain the resource_id that we would extract from the ARN (and we would have to know which dimension represents the resource_id).

So, I saw 3 options
1) The info metric depend on the tag selection feature usage (this initial PR)
2) The info metric is configured separately in a dedicated config bloc, so it's a way to both opt-in and remove dependency to the tag selection. Finally, this may be a way more intuitive solution (proposed in my [comment here](https://github.com/prometheus/cloudwatch_exporter/pull/229#issuecomment-551208385))
3) The info metric is transparently available to any metric selection, through an opt-in boolean. The problem here is, unless I miss something, we would have to add a dependency to each resource type SDK, and call and maintain specific code for each (e.g. building an `AmazonEC2` client, building a `DescribeTagsRequest()`, etc..) and even than, any correlation code with CloudWatch dimension would need to be hardcoded. I don't see it as an acceptable or maintainable solution.

Let me know what you think!

Reply:
> but the dependency to the tag selection is here because the necessary meta data is received through the

Ah yes.

> The info metric depend on the tag selection feature usage (this initial PR)

This would be the way to go I think, which means tag_selections should be optional.

Reply:
> This would be the way to go I think, which means tag_selections should be optional.

@brian-brazil . Good! So in summary:
- The tag selection mechanism (`aws_tag_select`) is already optional, so nothing to change here
- Do you still think we need a global only `enable_aws_resource_info` boolean, or should I just remove this boolean?

Reply:
> is already optional, so nothing to change here

The docs indicate otherwise.

> or should I just remove this boolean?

We can remove it.

Reply:
> The docs indicate otherwise.

The doc say "aws_tag_select | Optional. "
But you may refer to the tag_selection sub-element, which as changed in this PR because I added a coma.
The doc says: tag_selections | Required under `aws_tag_select`, maybe it's not clear enough, should I say instead:
"tag_selections | Required when using `aws_tag_select`"

Reply:
tag_selections should no longer be required, as it's not needed if you just want this new metric.

Reply:
Ah! yes ok.